sdk/attestation/azure-security-attestation/azure/security/attestation/_models.py

```diff
@@ -69,242 +82,248 @@ class PolicyCertificatesModificationResult(object):
     """
 
     def __init__(self, certificate_thumbprint, certificate_resolution):
-        #type:(str, CertificateModification)->None
+        # type: (str, CertificateModification, AttestationToken) -> None
         self.certificate_thumbprint = certificate_thumbprint
         self.certificate_resolution = certificate_resolution
 
     @classmethod
     def _from_generated(cls, generated):
+        # type: (GeneratedPolicyCertificatesModificationResult) -> AttestationPolicyCertificateResult
         if not generated:
             return cls
```

Comment:
If `None` is passed in you return a reference to the class, not an empty version of it. I think if `None` is passed in, `None` should be returned. Why are you returning `cls` here? (I'm assuming generated == None, is a weird edge case)

Reply:
Cargo-cult programming. I have no idea why generated would ever be `None`, but other _from_generated instances in `Tables` defended against that case using a similar pattern. I had a typo that it should have been `cls()` that I lost instead of `cls`.

But I have absolutely no idea why this pattern is required - I assumed it was expected for all `@classmethod`s.

Reply:
Let me start an email thread to see if there's collective guidance on the pattern to see which of the behaviors is canonical.